modules/ROOT/pages/logging-in-mule.adoc

```diff
@@ -0,0 +1,287 @@
+
+The output from the View Error Log is in the workspace's `.metadata/.log` file, for example:
+----
+/Users/me/AnypointStudio/workspace/.metadata/.log
+----
+
+After you run your Mule application, the log output is in the workspace's `.mule/logs/<project>.log` file.
+For example, for project `abcd`:
+
+----
+/Users/me/AnypointStudio/workspace/.mule/logs/abcd.log
+----
+
+== Synchronous Versus Asynchronous Logging
+
+[NOTE]
+By default, Mule logs messages asynchronously in version 3.6.0 and later.
+
+When logging synchronously, the execution of the thread that is processing your message is interrupted to wait for the log message to be fully handled before it can continue:
+
+image::logger-synch.jpg[logger+synch,align="center"]
+
+When logging asynchronously, the logging operation occurs in a separate thread, so the message can process before the logging completes:
+
+image::logger-asynch-alternativo.jpg[logger+asynch+alternativo,align="center"]
+
+In most situations, it's recommended that you use *asynchronous* logging, because it substantially improves the throughput and lowers the latency of message processing.
+
+When asynchronous logging is used, some actions may not be logged if there is a system crash. This occurs because log writing is performed on a separate thread that runs independently of other actions. See <<Exception Handling With Asynchronous Logging>> for information on mitigating this issue.
+
+If you use application logs as audit trails, configure your application to always use *synchronous* logging. This prevents the loss of any log messages.
+
+=== Performance tests
+The chart below shows the performance difference between synchronous and asynchronous logging, and how much latency increased as more concurrent messages were added. In this test, an application logged about one million messages, using an increasingly higher amount of threads on each run. Each transaction resulted in 1000 messages.
+
+image::logging-latency-vs-concurrency.png[]
+
+As indicated by the chart, the results of logging asynchronously are significantly closer to the results of not logging at all.
+
+== Configuring Custom Logging Settings
+
+By default, logging in Mule is done asynchronously and at a level greater than or equal to INFO. This discards log messages at the DEBUG or TRACE level.
+
+To use synchronous logging, adjust the logging level or define custom categories. You can configure these properties using the `$MULE_HOME/conf/log4j2.xml` file, which specifies how the logger behaves. If you don't edit this file, Mule uses the default properties.
+
+In Anypoint Studio, `log4j2.xml` appears in the `src/main/resources` path.
+
+The default configuration defines all loggers, including the root logger, as asynchronous. You can override this configuration at the domain or application level. To override this configuration at the application level, add a `logConfigFile` entry to the `mule-artiface,json` file. For example.
+
+---
```

Comment:
This is causing a broken style:
<img width="640" alt="Screen Shot 2019-03-25 at 4 56 09 PM" src="https://user-images.githubusercontent.com/16763154/54949965-00f2bf80-4f1f-11e9-92d8-f60519177db2.png">

If you are referencing a JSON sample, enclose it as your JSON:

```
[source,JSON,linenums]
----
{
  "minMuleVersion": "4.0.0",

   "logConfigFile": "../../../test-classes/resources/logging/custom-log4j2.xml"
}
----
```

Reply:
Fixed